packages/frontend/src/components/accounts/BatchImportAccounts.js

```diff
@@ -0,0 +1,444 @@
+        })),
+    });
+    const currentAccount = useMemo(() => state.accounts.find((account) => account.status === IMPORT_STATUS.PENDING), [state.accounts]);
+    const accountsApproved = useMemo(() => state.accounts.reduce((acc, curr) => curr.status === IMPORT_STATUS.SUCCESS ? acc + 1 : acc,0), [state.accounts]);
+    const completed = useMemo(() => state.accounts.every((account) => account.status === IMPORT_STATUS.SUCCESS || account.status === IMPORT_STATUS.FAILED), [state.accounts]);
+    const showSuccessScreen = useMemo(() => completed && state.accounts.some((account) => account.status === IMPORT_STATUS.SUCCESS), [completed, state.accounts]);
+
+    useEffect(() => {
+      if (!currentAccount) {
+        dispatch({type: ACTIONS.REMOVE_ACCOUNTS, accounts: availableAccounts});
+      }
+    },[availableAccounts]);
+
+    return showSuccessScreen ? (
+        <BatchImportAccountsSuccessScreen
+            accounts={state.accounts.filter(
+                ({ status }) => status === IMPORT_STATUS.SUCCESS
+            )}
+        />
+    ) : (
+        <>
+            <Container className="small-centered border ledger-theme">
+              <CustomContainer>
+                  <img src={ImportArrow} alt="ImportArrow" />
+                  <div className='screen-descripton'>
+                    <h3>
+                      <Translate id="batchImportAccounts.importScreen.title" data={{ noOfAccounts: state.accounts.length }}/>
+                      {accountUrlReferrer || <Translate id="sign.unknownApp" />}
+                    </h3>
+                    <br />
+                    <br />
+                    <Translate id="batchImportAccounts.importScreen.desc"/>
+                  </div>
+                    <div className="title">
+                        {accountsApproved}/{state.accounts.length}{' '}
+                        <Translate id="signInLedger.modal.accountsApproved" />
+                    </div>
+                    <AccountListImport accounts={state.accounts} />
+                    <div style={{ borderTop: '2px solid #f5f5f5' }} />
+                    <FormButtonGroup>
+                        <FormButton
+                            onClick={onCancel}
+                            className="gray-blue"
+                            disabled={availableAccountsIsLoading}
+                        >
+                            <Translate id="button.cancel" />
+                        </FormButton>
+                        <FormButton
+                            onClick={() =>
+                                dispatch({ type: ACTIONS.BEGIN_IMPORT })
+                            }
+                            disabled={availableAccountsIsLoading || completed}
+                        >
+                            <Translate id="button.beginImport" />
+                        </FormButton>
+                    </FormButtonGroup>
+                </CustomContainer>
+            </Container>
+            {currentAccount ? (
```

Comment:
on re-reviewing this we want the modal to not render at all if the current account doesn't exist but we still want to render a modal to prompt the user to confirm their url if the url is not confirmed. I think if we collapse it will still render the url confirm modal if no current account.